File: frontend/src/app/classes/task.ts
import { Example } from './example'
import { NullTask, TaskType } from './task-type'
import { Team } from './team'

export type TaskAndUsersWithProgress = {
  task: Task<Example>
  team: Team
}

export class Task<T extends Example> {
  public static readonly empty = new Task('', '', [], NullTask)
  public static deserialize(task: Task<Example>): Task<Example> {
    return new Task(
      task.id,
      task.name,
      task.examples,
      TaskType.deserialize(task.type),
      task.description)
  }

  public constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly examples: T[],
    public readonly type: TaskType,
    public readonly description?: string,
  ) { }
}

File: backend/src/main/java/pl/edu/ug/inf/lexica/domain/Task.java
package pl.edu.ug.inf.lexica.domain;

import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@RequiredArgsConstructor
@Entity
public class Task {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    private UUID id;

    @NonNull
    private String name;

    @NonNull
    @OneToMany(cascade = CascadeType.ALL)
    private List<Example> examples; //tu będzie musiało być List<Example>

    @NonNull
    private boolean isActive;

    @NonNull
    private String description;

    @NonNull
    @ManyToOne
    private TaskType type;
}
